### DISCUSSION TITLE
Federate module WITHIN NX repo for use in an EXTERNAL repo [Solved]

### DISCUSSION THREAD

**Discussion by davidsilva2841** [nrwl/nx#27720] [Questions]
Hi, I see there's some docs here about creating a remote federated module **_WITHIN_** a NX project for use in a host **_WITHIN_** a NX project. 
https://nx.dev/concepts/module-federation/module-federation-and-nx

Is there any support for federating a module **_WITHIN_** a NX repo for use in an **_EXTERNAL_** repo that **_does not use NX_**?

I tried using a federated module from a NX monorepo in an external repo that isn't using NX & am running into issues

Remote NX version: 19.6.4
Host Webpack version: 5.94.0

```
remoteEntry.js:1 Uncaught SyntaxError: Unexpected token 'export' (at remoteEntry.js:1:466702)

var __webpack_exports__ = __webpack_require__(630)
  , __webpack_exports__get = __webpack_exports__.get
  , __webpack_exports__init = __webpack_exports__.init;
export {__webpack_exports__get as get, __webpack_exports__init as init};
```

I've also gotten this error:
```
Loading script failed.
(missing: http://localhost:4201/remoteEntry.js)
while loading "./Button" from webpack/container/reference/app2
ScriptExternalLoadError
    at webpack/container/reference/app2 (http://localhost:3000/bundle.js:310:25)
    at __webpack_require__ (http://localhost:3000/bundle.js:350:32)
    at initExternal (http://localhost:3000/bundle.js:636:28)
    at __webpack_require__.I (http://localhost:3000/bundle.js:649:15)
    at http://localhost:3000/bundle.js:1146:47
    at webpack/sharing/consume/default/react/react (http://localhost:3000/bundle.js:1197:67)
    at http://localhost:3000/bundle.js:1231:56
    at Array.forEach (<anonymous>)
    at __webpack_require__.f.consumes (http://localhost:3000/bundle.js:1212:36)
    at http://localhost:3000/bundle.js:430:40
```

**Comment by davidsilva2841** [comment]
bump

**Comment by davidsilva2841** [comment]
I fixed this by specifying the library field here....
```
withModuleFederation(
  {
    // ...
    library: {
      type: 'var',
      name: 'app'
    },
    // ...
  }
);
```

**Comment by Coly010** [comment]
The export error is down to the host application.

It doesn’t really matter if the remote or host is in an Nx workspace or not, so long as it’s accessible.

If you have a remote that used by a host outside the Nx workspace, then as you’ve done already, you either need to configure the remote to be built to the same ESM/CJS artifact that the host is expecting.

within Nx workspaces we tend to stick to ESM, allowing modules and exports to be used.
Given your change, I’d hazard a guess that your host (outside the Nx workspace) is using CJS.

therefore your change makes sense. 

Just note that if you decide to consume that remote in a host inside the Nx workspace, you may need to either have a separate Webpack config that is used within Nx, or update your host within Nx to also use var
